packages.nix

```diff
@@ -32,10 +33,16 @@ let
 
       nativeCheckInputs = [
         clippy
+        nix
       ];
 
       preCheck = ''
         ${lib.getExe pkgs.cargo} clippy
+
+        # Stop the Nix command from trying to create /nix/var/nix/profiles.
+        #
+        # https://nix.dev/manual/nix/2.24/command-ref/new-cli/nix3-profile#profiles
+        export NIX_STATE_DIR=$TMPDIR
```

Comment:
Is this here only to make tests not fail?

Reply:
Ah, just read through your above comments. Makes sense.

Reply:
Maybe it would make more sense to have this exported by the test thread instead of as part of the package? Going to try it out really quick.

Reply:
Ah well. This is just part of the check phase anyway.